src/ringmesh/io/geomodel_adapter_resqml.cpp

```diff
@@ -0,0 +1,687 @@
+                    return false;
+                }
+            }
+        }
+
+        for( const auto& interface :
+            geomodel_.geol_entities( Interface3D::type_name_static() ) )
+        {
+            GMGE::GEOL_FEATURE geo_feat = interface.geological_feature();
+            if( !interface.has_geological_feature()
+                || !( GMGE::is_fault( geo_feat )
+                       || GMGE::is_stratigraphic_limit( geo_feat )
+                       || GMGE::GEOL_FEATURE::VOI == geo_feat ) )
+            {
+                Logger::warn( "", interface.gmge(),
+                    " geological feature is empty or not recognized" );
+            }
+
+            AbstractFeature* feature = find_or_create_feature( interface );
+            AbstractFeatureInterpretation* interp =
+                find_or_create_interpretation( *feature );
+
+            std::string guid( GuidTools::generateUidAsString() );
+            TriangulatedSetRepresentation* rep =
+                pck_->createTriangulatedSetRepresentation(
+                    interp, local_3d_crs_, guid, feature->getTitle() );
+
+            const index_t interface_vertex_count =
+                write_triangulated_patches( interface, rep );
+
+            if( interface_vertex_count == 0 )
+            {
+                return true;
+            }
+
+            std::vector< AbstractValuesProperty* > vertex_properties;
+            std::vector< AbstractValuesProperty* > polygon_properties;
+            for( auto i : range( interface.nb_children() ) )
+            {
+                const Surface3D& surface =
+                    static_cast< const Surface3D& >( interface.child( i ) );
+
+                write_property( i == 0, vertex_properties,
+                    surface.vertex_attribute_manager(),
+                    gsoap_resqml2_0_1::resqml2__IndexableElements__nodes,
+                    *rep );
+                write_property( i == 0, polygon_properties,
+                    surface.polygon_attribute_manager(),
+                    gsoap_resqml2_0_1::resqml2__IndexableElements__cells,
+                    *rep );
+            }
+        }
+        return true;
+    }
+
+    bool GeoModelAdapterRESQMLImpl::write_volumes()
+    {
+        for( const auto& layer :
+            geomodel_.geol_entities( Layer3D::type_name_static() ) )
+        {
+            AbstractFeatureInterpretation* interp = nullptr;
+            if( layer.has_geological_feature() )
+            {
+                AbstractFeature* feature = find_or_create_feature( layer );
+                AbstractFeatureInterpretation* interp =
+                    find_or_create_interpretation( *feature );
+            }
+            else
+            {
+                Logger::warn( "", layer.gmge(),
+                    " geological feature is empty or not recognized" );
+            }
+
+            // TODO: I do not understand why the interp is not needed...
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1135)